proxy/examples_test.go

```diff
@@ -19,8 +22,10 @@ var director proxy.StreamDirector
 
 // ExampleRegisterService is a simple example of registering a service with the proxy.
 func ExampleRegisterService() {
+	encoding.RegisterCodec(proxy.Codec())
```

Comment:
On further investigation I believe we can remove this call, since we require callers to explicitly set our codec. AKA we don't want to override the default one. And both server and client created with `grpc.Force*` call should do the trick.

Reply:
Nice find for `grpc.ForceServerCodec`! Not a fan of the global state either